modules/http/src/main/scala/ch/epfl/bluebrain/nexus/commons/http/PathSanity.scala

```diff
@@ -0,0 +1,69 @@
+package ch.epfl.bluebrain.nexus.commons.http
+
+import akka.http.scaladsl.model.Uri.Path
+import ch.epfl.bluebrain.nexus.commons.http.PathSanity.Position._
+
+import scala.annotation.tailrec
+
+object PathSanity {
```

Comment:
I think this logic should be added to the Path itself... It's strongly coupled to Path.

Reply:
Come to think of it, the Path should automatically collapse duplicate slashes and split newly added segments into multiple segments if necessary..